apollo-router-core/src/query_planner/mod.rs

```diff
@@ -253,66 +270,129 @@ impl PlanNode {
     }
 }
 
-/// A fetch node.
-#[derive(Debug, PartialEq, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub(crate) struct FetchNode {
-    /// The name of the service or subgraph that the fetch is querying.
-    service_name: String,
+mod fetch {
+    use std::sync::Arc;
 
-    /// The data that is required for the subgraph fetch.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    requires: Option<Vec<Selection>>,
+    use super::selection::{select_object, Selection};
+    use futures::StreamExt;
```

Comment:
:yarn: less important management this way
```suggestion
    use futures::prelude::*;
```